### PR TITLE
Feat/image loading improvement

### LINKED ISSUE #26: US12 - Image Loading Improvement
## Value proposition

As an user
I want to see multiple pages in the breed list
In order to have a better performance of the app

## Description

Bildschirmfoto 2023-08-29 um 10 51 47

## Acceptance criteria

- [x] On the breed list page I can only see twenty entries at a time
- [x] On the bottom there are two buttons to swap pages and show the next / prev entries
- [x] ALTERNATIV - there is a load more button, that loads the next 20 on the bottom of the current list
- [x] when new Entries are loading I can see a loading state

## Tasks

- [x] `create a branch feat/image-loading-improvement`
- [x] split the given breeds data array in smaller ones via slice
- [x] adjust the start / end index when clicking on the buttons next / prev
- [x] add a loading state for the entries

### PR DESCRIPTION
**Pull request by ninaassmann**
- the breeds list shows now only ten at a time
- there are two buttons on the bottom to switch "pages"
- added a function to track the sliceOptions (I used slice on the array that shows the breeds)
- prev / next button are diabled when the user gets to first or the last breed on the list
- when filtering the breeds I reset the sliceOptions in order to fixed the issue that I used jumps of ten

I put everything in the localstorage so the user does not have to filter again, when coming back to this page
that works for the filter, but not for the pages. When i go to page two, click on one breed and then go back, I got back on page one.

### CODE REVIEW

**Comment by ninaassmann** [comment]
fixed the issue from the discussion board myself

File: components/Breeds/Thumbnail.js [hunk 1]
```diff
@@ -1,37 +1,57 @@
-import Image from "next/image";
-import { styled } from "styled-components";
 import useSWR from "swr";
+import StyledImage from "../Image.styled";
+import ImageWrapper from "../ImageWrapper.styled";
+import { keyframes, styled } from "styled-components";
+import { useState } from "react";
 
 export default function Thumbnail({ breed }) {
+  const [loader, setLoader] = useState(true);
```
Comment (MerleMak) [inline components/Breeds/Thumbnail.js]: I see why you use a useState here, but technically the state "true" or "false" are also saved in the information of useSWR if the data is fetched yet. You see,  the "isLoading" from useSWR also indicates a boolean. So you could avoid this useState all together and use the isLoading from useSWR instead.
```suggestion
```

File: components/Breeds/Thumbnail.js [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
+
   const { data, isLoading, error } = useSWR(
-    `/api/dogBreeds/${breed.reference_image_id}`
+    `/api/dogBreeds/${breed && breed.reference_image_id}`
   );
   if (error) return <div>failed to load</div>;
   if (isLoading) return <div>loading...</div>;
```
Comment (MerleMak) [inline components/Breeds/Thumbnail.js]:
```suggestion
  if (isLoading) return (
  <LoaderWrapper>
          <Loader />
  </LoaderWrapper>
   );
```

File: components/Breeds/Thumbnail.js [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
-        alt={breed.name}
-        width={data.results.width}
-        height={data.results.height}
-      />
+    <ImageWrapper $variant="thumbnail">
+      {loader && (
+        <LoaderWrapper>
+          <Loader />
+        </LoaderWrapper>
+      )}
```
Comment (MerleMak) [inline components/Breeds/Thumbnail.js]:
```suggestion
```

File: pages/breeds/index.js [hunk 4]
```diff
@@ -10,21 +9,46 @@ import {
...
+  });
+  const [sliceOptions, setSliceOptions] = useLocalStorageState("sliceOptions", {
+    defaultValue: initialSliceOptions,
+  });
 
-  const breedsToShow = dogBreedFilter(filter, dogBreeds);
+  const breedsToShow = dogBreedFilter(filter, dogBreeds && dogBreeds);
+  const breedsToShowCount = breedsToShow?.length;
+
+  function handleOnChange(event, key) {
```
Comment (MerleMak) [inline pages/breeds/index.js:44]: nice 👌🏻

**Review by MerleMak — COMMENTED** [review]
Really really nice job! I left you a comment for the useState in the Thumbnail :) keep it up!!

File: components/Breeds/Thumbnail.js (hunk 1 above)
Comment (ninaassmann) [reply components/Breeds/Thumbnail.js]: already had something like that in my mind, thanks for the push

**Review by izabelladobielewska — APPROVED** [review]
LGTM:)

**Pull request merged**